zio-http/src/test/scala/zhttp/service/ClientSpec.scala

```diff
@@ -1,16 +1,48 @@
 package zhttp.service
 
-import zhttp.internal.HttpRunnableSpec
-import zhttp.service.client.ClientSSLHandler.ClientSSLOptions
-import zio.test.Assertion.anything
-import zio.test.assertM
+import zhttp.http._
+import zhttp.internal.{AppCollection, HttpRunnableSpec}
+import zhttp.service.server._
+import zio.duration.durationInt
+import zio.test.Assertion._
+import zio.test.TestAspect._
+import zio.test._
 
 object ClientSpec extends HttpRunnableSpec(8082) {
-  val env           = ChannelFactory.auto ++ EventLoopGroup.auto()
-  override def spec = suite("Client")(
+
+  override def spec = {
+    suiteM("Client") {
+      serve(AppCollection.app).as(List(clientSpec)).useNow
+    }.provideCustomLayerShared(env) @@ timeout(5 seconds) @@ sequential
+  }
+  def clientSpec    = suite("ClientSpec") {
     testM("respond Ok") {
-      val actual = Client.request("http://api.github.com/users/zio/repos", ClientSSLOptions.DefaultSSL)
+      val actual = Http.ok.requestBodyAsString()
       assertM(actual)(anything)
-    },
-  ).provideCustomLayer(env)
+    } +
+      testM("non empty content") {
+        val actual          = Http.response(Response(status = Status.OK, data = HttpData.fromText("abc")))
+        val responseContent = actual.requestBody()
+        assertM(responseContent)(isNonEmpty)
+      } +
+      testM("POST request expect non empty response content") {
+        val headers  = Headers.userAgent("zio-http test")
```

Comment:
why are headers important in this test?

Reply:
it is just a left out from previous version, where the initial call was made externally to Github and GitHub API requires to provide a valid user -agent  header. Maybe we should set up one by default in the zio-http client. For this change I will remove it and improve the testing.